watertap/unit_models/translator_adm1_asm1.py

```diff
@@ -0,0 +1,438 @@
+                blk.properties_out[t].conc_mass_comp["X_I"]
+                == blk.properties_in[t].conc_mass_comp["X_I"]
+            )
+
+        self.readily_biodegradable = Set(
+            initialize=["S_su", "S_aa", "S_fa", "S_va", "S_bu", "S_pro", "S_ac"]
+        )
+
+        self.slowly_biodegradable = Set(
+            initialize=[
+                "X_c",
+                "X_ch",
+                "X_pr",
+                "X_li",
+                "X_su",
+                "X_aa",
+                "X_fa",
+                "X_c4",
+                "X_pro",
+                "X_ac",
+                "X_h2",
+            ]
+        )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality S_S equation",
+        )
+        def eq_SS_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_S"] == sum(
+                blk.properties_in[t].conc_mass_comp[i]
+                for i in blk.readily_biodegradable
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality X_S equation",
+        )
+        def eq_XS_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["X_S"] == sum(
+                blk.properties_in[t].conc_mass_comp[i] for i in blk.slowly_biodegradable
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality S_NH equation",
+        )
+        def eq_Snh_conc(blk, t):
+            return (
+                blk.properties_out[t].conc_mass_comp["S_NH"]
+                == blk.properties_in[t].conc_mass_comp["S_IN"]
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality S_ND equation",
+        )
+        def eq_Snd_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_ND"] == (
+                blk.properties_in[t].conc_mass_comp["S_I"] * blk.N_I
+            ) + (blk.properties_in[t].conc_mass_comp["S_I"] * blk.N_aa)
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Equality Xnd equation",
+        )
+        def eq_Xnd_conc(blk, t):
+            return blk.properties_out[t].conc_mass_comp["X_ND"] == (
+                (
+                    blk.N_bac
+                    * (
+                        blk.properties_in[t].conc_mass_comp["X_su"]
+                        + blk.properties_in[t].conc_mass_comp["X_aa"]
+                        + blk.properties_in[t].conc_mass_comp["X_fa"]
+                        + blk.properties_in[t].conc_mass_comp["X_c4"]
+                        + blk.properties_in[t].conc_mass_comp["X_pro"]
+                        + blk.properties_in[t].conc_mass_comp["X_ac"]
+                        + blk.properties_in[t].conc_mass_comp["X_h2"]
+                    )
+                )
+                + (blk.properties_in[t].conc_mass_comp["X_I"] * blk.N_I)
+                + (blk.properties_in[t].conc_mass_comp["X_c"] * blk.N_I)
```

Comment:
Is there a reason `N_I` is used here over the `N_xc` parameter?

Reply:
Changed. However, we should evaluate if this parameters live here